### PR TITLE
Docs(repo): Introduce Spirit release schedule document

### PR DESCRIPTION
**Pull request by literat**
## Description

Introducing a document with the description of our release schedule and some notes on what this means for users/developers.

### Additional context

Some other questions for consideration:

- Do we want to include a link to our quarterly goals or somehow describe what big changes are we planning in the next major release?
- How to communicate that the `web-twig` will be abandoned in the next major release? Should it be part of this document (applies to the first question also)?

### Issue reference

https://jira.almacareer.tech/browse/DS-1552

### CODE REVIEW

File: docs/contribution/release-schedule.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+| v1 🌟 Aura   | End of Life | 2023-07-21      | 2023-07-21   | 2024-06-10        | 2024-10-17  |
+| v2 🍃 Breeze | Maintenance | 2024-06-10      | 2024-06-10   | 2024-11-25        | -/-         |
+| v3 ✨ Charm  | Active      | 2024-10-17      | 2024-11-25   | -/-               | -/-         |
+| v4 🌈 Divine | Unreleased  | -/-             | -/-          | -/-               | -/-         |
+
+> Dates are subject to change
+
+![schedule][release-schedule-static]
+
+```
```
Comment (literat) [inline docs/contribution/release-schedule.md:19]: **question (blocking):** Do we want the schedule in textual view or rather as an image?

File: static/release-schedule.svg
Comment (literat) [inline static/release-schedule.svg:1]: **question (if-minor):** Do we want to use our brand colors in the image?

File: docs/contribution/release-schedule.md (hunk 1 above)
Comment (literat) [reply docs/contribution/release-schedule.md:19]: Image generated by https://github.com/lmc-eu/spirit-design-system/pull/1792

File: docs/contribution/release-schedule.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+## Release Phases
+
+### Prerelease
+
+The prerelease phase is intended to be the opportunity for early adopters,
+library authors, and other strategic ecosystem partners to begin to evaluate and
+integrate new changes into their codebases. For v3, this phase was more than a one month
+long and spanned five prerelease/alpha releases. We hope to extend this timeframe
+even further for our next major.
```
Comment (adamkudrna) [inline docs/contribution/release-schedule.md:46]:
```suggestion
even further for our next major version.
```

File: docs/contribution/release-schedule.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+### Active
+
+Consuming projects should always aim to follow the Active release.
+
+A release in the Active phase receives biweekly minor releases containing new
+features and fixes. The work we deliver into `main` every day is considered
+unstable. Almost every two weeks we package up these changes into a new minor version
+that is published from `main` to the current Active major.
```
Comment (adamkudrna) [inline docs/contribution/release-schedule.md:55]:
```suggestion
that is published from `main` to the current Active major version.
```

🤔

File: docs/contribution/release-schedule.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+security patches and critical bug fixes. When a version moves from Active to
+Maintenance, consuming projects should begin migrating to the new Active major
+version. During Maintenance we also consider adding non-critical bug fixes on an
+ad hoc basis, by request only. To request a fix be back-ported to a Maintenance
+release, please
+[open an issue][spirit-jira-board] or let us know in [support Slack channel][spirit-support-slack-channel].
+
+Changes required for critical security and bug fixes may lead to semver-major
+changes landing within a release stream, such situations will be rare and will
+land as semver-minor. Although, those changes should have a revert option
```
Comment (adamkudrna) [inline docs/contribution/release-schedule.md:69]: > will land as semver-minor

I understand (or I hope I do), but do we want to do that…? Does it mean a security fix may require changes in our users' code base?

File: docs/contribution/release-schedule.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+major versions of the Spirit Design System.
+
+| Release      | Status      | Initial release | Begin Active | Begin Maintenance | End of Life |
+| ------------ | ----------- | --------------- | ------------ | ----------------- | ----------- |
+| `main`       | unstable    | unstable        | unstable     | unstable          | unstable    |
+| v0           | End of Life | 2021-09-20      | 2021-09-20   | 2023-07-21        | 2024-06-10  |
+| v1 🌟 Aura   | End of Life | 2023-07-21      | 2023-07-21   | 2024-06-10        | 2024-10-17  |
+| v2 🍃 Breeze | Maintenance | 2024-06-10      | 2024-06-10   | 2024-11-25        | -/-         |
+| v3 ✨ Charm  | Active      | 2024-10-17      | 2024-11-25   | -/-               | -/-         |
+| v4 🌈 Divine | Unreleased  | -/-             | -/-          | -/-               | -/-         |
```
Comment (adamkudrna) [inline docs/contribution/release-schedule.md:13]: ❤️

File: docs/contribution/release-schedule.md
Comment (adamkudrna) [inline docs/contribution/release-schedule.md:1]: It's not explicitly mentioned anywhere that we only provide support for the Active and Maintenance life phases. I'd consider adding that, what do you think?

Comment (adamkudrna) [inline docs/contribution/release-schedule.md:1]: >  Do we want to include a link to our quarterly goals or somehow describe what big changes are we planning in the next major release?

Yes, I would add the link to Notion right after the JIRA Release Board link.

> How to communicate that the web-twig will be abandoned in the next major release? Should it be part of this document (applies to the first question also)?

I think this document is rather of general nature. Quarterly goals can change and the less places to maintain, the better. But I am open to discussion.

**Review by adamkudrna — APPROVED** [review]
Wow, well done! 👏🏻

File: static/release-schedule.svg
Comment (literat) [reply static/release-schedule.svg:1]: Brand colors used.

File: docs/contribution/release-schedule.md (hunk 1 above)
Comment (literat) [reply docs/contribution/release-schedule.md:19]: OK.

File: docs/contribution/release-schedule.md (hunk 3 above)
Comment (literat) [reply docs/contribution/release-schedule.md:55]: Fixed.

File: docs/contribution/release-schedule.md (hunk 4 above)
Comment (literat) [reply docs/contribution/release-schedule.md:69]: I think if that would be so critical and crucial it should be that way. We have a process for BC that requires a lot of work and documents to be created. So we have to have an exception fot this process. And shit sometimes happen 🤷‍♂️

File: docs/contribution/release-schedule.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+repositories.
+
+## Acknowledgements
+
+This document was heavily inspired by the work of the
+[NodeJS Release Working Group][node-js-release-working-group] and [Carbon Design System Release Schedule][carbon-release-schedule].
+
+[carbon-release-schedule]: https://github.com/carbon-design-system/carbon/blob/main/docs/release-schedule.md
+[node-js-release-working-group]: https://github.com/nodejs/release
+[release-schedule-static]: ../../static/release-schedule.svg
```
[the remark this answers is not in the document]
Comment (literat) [reply docs/contribution/release-schedule.md:84]: Fixed.

File: docs/contribution/release-schedule.md
Comment (literat) [reply docs/contribution/release-schedule.md:1]: I have added another section with that.

**Pull request merged**